src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_fullyconnected_primitive.cpp

```diff
@@ -0,0 +1,356 @@
+    auto first_desc = dnnl::inner_product_forward::primitive_desc(prim_desc.get());
+
+    const bool found = DnnlExtensionUtils::find_implementation(prim_desc, [&](impl_desc_type implType) {
+        return contains(implPriorities, implType);
+    });
+
+    if (found)
+        return std::move(prim_desc);
+
+    return std::move(first_desc);
+}
+
+static VectorDims makeDummyInputDims(const Shape& inShape, const Shape& wShape) {
+    const auto& weightDims = wShape.getStaticDims();
+
+    auto inMinDims = inShape.getMinDims();
+    auto inMaxDims = inShape.getMaxDims();
+    inMinDims.back() = weightDims.back();
+    inMaxDims.back() = weightDims.back();
+
+    return MemoryDescUtils::makeDummyShape(Shape(inMinDims, inMaxDims)).getStaticDims();
+}
+
+static VectorDims makeDummyOutputDims(const VectorDims& inShape, const VectorDims& wShape, const size_t out_rank) {
+    size_t activationRank = inShape.size();
+    size_t channelRank = wShape.size() - 1;
+    // activation   weight    output_shape
+    // NCHW         CoCHW     NCo
+    // TNC          CoC       TNCo
+    // NC           CoC       NCo
+    VectorDims outputShape(out_rank, 1);
+    // set Co
+    outputShape.back() = wShape[0];
+    // set batch dims
+    size_t batchRank = activationRank - channelRank;
+    size_t startIdx = out_rank - batchRank - 1;
+    for (size_t i = 0; i < batchRank; i++) {
+        outputShape[i + startIdx] = inShape[i];
+    }
+
+    return outputShape;
+}
+
+DnnlShapeAgnosticDataPtr DnnlFCPrimitive::createShapeAgnosticData(const FCAttrs& attrs,
+                                                                  const PostOps& postOps,
+                                                                  const MemoryArgs& memory,
+                                                                  const ExecutorContext::CPtr context,
+                                                                  const bool cacheWeights) {
+    DEBUG_LOG("Creating shape agnostic data");
+    auto srcDesc = memory.at(ARG_SRC)->getDescPtr();
+    const auto& weiDesc = memory.at(ARG_WEI)->getDescPtr();
+    auto dstDesc = memory.at(ARG_DST)->getDescPtr();
+
+    if (srcDesc->getShape().isDynamic()) {
+        const auto& inShape = srcDesc->getShape();
+        const auto& wShape = weiDesc->getShape();
+        const auto& inDymmyDims = makeDummyInputDims(inShape, wShape);
+        srcDesc = srcDesc->cloneWithNewDims(inDymmyDims);
+        const auto& outDymmyDims =
+            makeDummyOutputDims(inDymmyDims, wShape.getStaticDims(), dstDesc->getShape().getRank());
+        dstDesc = dstDesc->cloneWithNewDims(outDymmyDims);
+    }
+
+    const dnnl::memory::desc srcDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(srcDesc)->getDnnlDesc();
+    const dnnl::memory::desc weiDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(weiDesc)->getDnnlDesc();
+    const dnnl::memory::desc dstDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(dstDesc)->getDnnlDesc();
```

Comment:
Shouldn't we create Dnnl descs after `if (!cacheWeights)` condition?

Reply:
Done